pkg/crossplane/instance.go

```diff
@@ -151,3 +155,58 @@ func (cp *Crossplane) GetInstance(ctx context.Context, instanceID string) (*comp
 	}
 	return nil, ErrInstanceNotFound
 }
+
+// UpdateInstanceSLA updates the SLA of an instance specified by the supplied planID.
+// Only SLA changes are allowed, any other change is not permitted and yields an error.
+func (cp *Crossplane) UpdateInstanceSLA(ctx context.Context, instanceID, serviceID, planID string) error {
+	instance, err := cp.GetInstance(ctx, instanceID)
+	if err != nil {
+		return err
+	}
+
+	instanceLabels := instance.GetLabels()
+	if serviceID != instanceLabels[ServiceIDLabel] {
+		return ErrServiceUpdateNotPermitted
+	}
+
+	newPlan, err := cp.GetPlan(ctx, planID)
+	if err != nil {
+		return err
+	}
+
+	instanceSLA := instanceLabels[SLALabel]
+	newPlanSLA := newPlan.Labels[SLALabel]
+
+	slaChangePermitted := func() bool {
```

Comment:
As far as I understand, this doesn't catch the case where for example `medium` will be updated to `large-premium`?

Reply:
exactly - wasn't specified exactly in the ticket therefore I wasn't sure if it's required. Question is how to check? string substitution is doable but not very pretty :)

Reply:
I fear it's gonna be string checking for now, yes :disappointed: